python/test/Test_dryad_TabulatedCrossSection.py

```diff
@@ -0,0 +1,1571 @@
+            self.assertAlmostEqual( 4.              , linear.values[2] )
+            self.assertAlmostEqual( 3.85048128530886, linear.values[3] )
+            self.assertAlmostEqual( 3.70951129135145, linear.values[4] )
+            self.assertAlmostEqual( 3.44966028678679, linear.values[5] )
+            self.assertAlmostEqual( 3.21459646033567, linear.values[6] )
+            self.assertAlmostEqual( 3.              , linear.values[7] )
+            self.assertAlmostEqual( 2.72176678584324, linear.values[8] )
+            self.assertAlmostEqual( 2.46416306545103, linear.values[9] )
+            self.assertAlmostEqual( 2.22433973930853, linear.values[10] )
+            self.assertAlmostEqual( 2.              , linear.values[11] )
+
+            self.assertEqual( True, linear.is_linearised )
+
+        def verify_chunk5( self, chunk ) :
+
+            # verify content
+            self.assertEqual( 5, len( chunk.energies ) )
+            self.assertEqual( 5, len( chunk.values ) )
+            self.assertEqual( 2, len( chunk.boundaries ) )
+            self.assertEqual( 2, len( chunk.interpolants ) )
+            self.assertAlmostEqual( 1., chunk.energies[0] )
+            self.assertAlmostEqual( 2., chunk.energies[1] )
+            self.assertAlmostEqual( 2., chunk.energies[2] )
+            self.assertAlmostEqual( 3., chunk.energies[3] )
+            self.assertAlmostEqual( 4., chunk.energies[4] )
+            self.assertAlmostEqual( 4., chunk.values[0] )
+            self.assertAlmostEqual( 3., chunk.values[1] )
+            self.assertAlmostEqual( 4., chunk.values[2] )
+            self.assertAlmostEqual( 3., chunk.values[3] )
+            self.assertAlmostEqual( 2., chunk.values[4] )
+            self.assertEqual( 1, chunk.boundaries[0] )    # <-- this is changed from 2 to 1
+            self.assertEqual( 4, chunk.boundaries[1] )
+            self.assertEqual( InterpolationType.LinearLinear, chunk.interpolants[0] )
+            self.assertEqual( InterpolationType.LinearLog, chunk.interpolants[1] )
+            self.assertEqual( False, chunk.is_linearised )
+
+        def verify_chunk6( self, chunk ) :
+
+            # verify content
+            self.assertEqual( 4, len( chunk.energies ) )
+            self.assertEqual( 4, len( chunk.values ) )
+            self.assertEqual( 2, len( chunk.boundaries ) )
+            self.assertEqual( 2, len( chunk.interpolants ) )
+            self.assertAlmostEqual( 1., chunk.energies[0] )
+            self.assertAlmostEqual( 2., chunk.energies[1] )
+            self.assertAlmostEqual( 3., chunk.energies[2] )
+            self.assertAlmostEqual( 4., chunk.energies[3] )      # <-- last point removed
+            self.assertAlmostEqual( 4., chunk.values[0] )
+            self.assertAlmostEqual( 3., chunk.values[1] )
+            self.assertAlmostEqual( 2., chunk.values[2] )
+            self.assertAlmostEqual( 1., chunk.values[3] )      # <-- last point removed
```

Comment:
Are you sure you want this to behave this way?
When not specified, MCNP goes ahead and assumes extrapolations (typically nearest-neighbor for above and below, but below has a few exceptions). This would remove evaluator control on this.

Reply:
I specifically do this when the value goes to zero at the end of the data. There is a total cross section (I think it is total) in one or two evaluations that does this while the other cross sections don;t. It feels like a manipulation error rather than a deliberate evaluation choice.